compiler/rustc_resolve/src/imports.rs

```diff
@@ -327,6 +328,15 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                         } else if !old_binding.vis.is_at_least(binding.vis, this.tcx) {
                             // We are glob-importing the same item but with greater visibility.
                             resolution.binding = Some(binding);
+                        } else if binding.is_ambiguity() && !binding.warning_ambiguity {
+                            resolution.binding =
+                                Some(self.arenas.alloc_name_binding(NameBinding {
+                                    ambiguity: Some((binding, AmbiguityKind::GlobVsGlob)),
+                                    warning_ambiguity: true,
```

Comment:
Looks like `warning_ambiguity` should be set to true in all cases where `old_binding.is_some()`, not just here.

Reply:
I don't understand why it's related to `old_binding`, can you explain it in detail?

Reply:
Because the update mechanism fix is responsible for majority of the regressions, not new glob-vs-glob ambiguities added here.